frontend/src/routes/decrypt/+page.svelte

```diff
@@ -11,92 +12,173 @@
   import { onMount } from "svelte";
   import Auth from "./auth.svelte";
   import Password from "./password.svelte";
+  import { getError, hasError } from "$lib/form";
+  import { AlertCircle } from "lucide-svelte";
 
   onMount(() => {
     renderLoginButton(document.getElementById("login-button"));
   });
 
-  let encodedPayload = "";
-  let plainText: Promise<string> | null = null;
-  let error: any = null;
-  let decrypt: () => Promise<string>;
+  const DecryptFormSchema = z.object({
+    encodedPayload: z.string().min(1),
+    payload: z
+      .custom<Cipherly.Payload | null>()
+      .refine((payload) => {
+        // version validation
```

Comment:
Custom validation logic can be written her against the payload.
returning false when invalid, along with a message.

Failed validation will set the `validationError` accordingly, then an error message is rendered below.

Reply:
Noice! I don't think we need to validate `payload` – we should only validate `encodedPayload` because `payload` is an intermediate step, and is not provided by the user. 

Perhaps it doesn't even belong in the DecryptFormSchema? Not sure if it needs to be in the schema for other reasons.